Update nws.markdown to match current behavior

The00Dustin opened the pull request:
## Proposed change
Update documentation to reflect my understanding of the latest changes mentioned in the "Backward-incompatible changes" section of the 2024.4 release notes.

## Type of change
- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [x] Removed stale or deprecated documentation.

## Additional information
- Link to parent pull request in the codebase: [#112503](https://github.com/home-assistant/core/pull/112503)
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #29256

## Checklist
- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

The00Dustin committed:
Update nws.markdown to current behavior

The00Dustin committed:
Add link to related documentation

MatthewFlamm commented on source/_integrations/nws.markdown:
```diff
@@ -25,6 +25,8 @@ According to the [API documentation](https://www.weather.gov/documentation/servi
 Providing a METAR station code is optional, and if not supplied, the closest station to the latitude and longitude will be chosen. A list of nearby stations is printed to the log with level `DEBUG` if no station is supplied. Stations can also be found on the [NOAA website](https://www.cnrfc.noaa.gov/metar.php). Codes with only three characters, for example, `ADW` should be prefixed with the lette ...
 
-Two weather entities are created for each entry in the configuration: one for day and night forecasts and one for hourly forecasts. The hourly forecast entity is disabled after configuration but can be enabled by the user. The time supplied for each forecast is the start time for the forecast. Sensors are also created as disabled entities after configuration and can be enabled by the user.
+One weather entity is created for each entry in the configuration for day and night forecasts. The time supplied for each forecast is the start time for the forecast. Hourly forecast data can be obtained using the `weather.get_forecasts` service. Sensors are also created as disabled entities after configuration and can be enabled by the user.
```
The way you have worded this makes it sound like the entity is created for day/night forecast and the hourly forecast needs to be obtained via `get_forecasts`.  In reality, one entity is for both forecasts and `get_forecasts` can be used for either type.

The00Dustin replied:
Should I change "day and night" to "day/night" or do you have another suggestion on how to clarify this?  Perhaps I need to read the latest change info and day/night isn't even accurate anymore, but that's the one entity I got when I installed in October...

The00Dustin replied:
That verbiage was original verbiage that I didn't touch.  After reviewing the release notes, I have modified that language and added a commit.  I am marking this concern as resolved accordingly.

The00Dustin committed:
Clarify weather entity purpose

The old language used "day and night" but had context to make it clear this was a single entity.  With the "hourly" verbiage removed, this change should reduce confusion.

MatthewFlamm commented on source/_integrations/nws.markdown:
```diff
@@ -25,6 +25,8 @@ According to the [API documentation](https://www.weather.gov/documentation/servi
 Providing a METAR station code is optional, and if not supplied, the closest station to the latitude and longitude will be chosen. A list of nearby stations is printed to the log with level `DEBUG` if no station is supplied. Stations can also be found on the [NOAA website](https://www.cnrfc.noaa.gov/metar.php). Codes with only three characters, for example, `ADW` should be prefixed with the lette ...
 
-Two weather entities are created for each entry in the configuration: one for day and night forecasts and one for hourly forecasts. The hourly forecast entity is disabled after configuration but can be enabled by the user. The time supplied for each forecast is the start time for the forecast. Sensors are also created as disabled entities after configuration and can be enabled by the user.
+One weather entity is created to provide forecasts for each entry in the configuration. The time supplied for each forecast is the start time for the forecast. Hourly forecast data can be obtained using the `weather.get_forecasts` service. Sensors are also created as disabled entities after configuration and can be enabled by the user.
```
```suggestion
One weather entity is created for each entry in the configuration with hourly and day/night forecasts, which can be obtained through the `weather.get_forecasts` service. The time supplied for each forecast is the start time for the forecast. Sensors are also created as disabled entities after configuration and can be enabled by the user.
```

The00Dustin replied:
I can commit that if you especially like it, but "with" threw me off briefly, so it might make sense to replace the word "with" using the word "providing".   I'm not sure that comma is grammatically correct, either, so maybe it should be broken into more sentences like this?:
> One weather entity is created for each entry in the configuration.  This entity provides hourly and day/night forecasts which can be viewed in the UI or accessed using the `weather.get_forecasts` service.

Even if that's better, I'm not sure if I should say "the UI" or "lovelace" though, so let me know what you think.

MatthewFlamm commented on source/_integrations/nws.markdown:
```diff
@@ -25,6 +25,8 @@ According to the [API documentation](https://www.weather.gov/documentation/servi
 Providing a METAR station code is optional, and if not supplied, the closest station to the latitude and longitude will be chosen. A list of nearby stations is printed to the log with level `DEBUG` if no station is supplied. Stations can also be found on the [NOAA website](https://www.cnrfc.noaa.gov/metar.php). Codes with only three characters, for example, `ADW` should be prefixed with the lette ...
 
-Two weather entities are created for each entry in the configuration: one for day and night forecasts and one for hourly forecasts. The hourly forecast entity is disabled after configuration but can be enabled by the user. The time supplied for each forecast is the start time for the forecast. Sensors are also created as disabled entities after configuration and can be enabled by the user.
+One weather entity is created to provide forecasts for each entry in the configuration. The time supplied for each forecast is the start time for the forecast. Hourly forecast data can be obtained using the `weather.get_forecasts` service. Sensors are also created as disabled entities after configuration and can be enabled by the user.
```
What about this:

```suggestion
One weather entity is created for each entry in the configuration. Hourly and day/night forecasts are provided through the `weather.get_forecasts` service. The time supplied for each forecast is the start time for the forecast. Sensors are also created as disabled entities after configuration and can be enabled by the user.

```

The00Dustin replied:
Technically sounds great, and I'm happy to commit it if you don't want to mention UI visibility.  The reason I mentioned UI visibility is because it might not be implicitly obvious to new users that the default dashboard and entity display already use this service, and that lack of knowledge might deter them from choosing this integration over others.

MatthewFlamm replied:
I actually dont know what you mean by this.  This integration to my knowledge is no different than other weather integrations.

The00Dustin replied:
Other documentation I looked at was less thorough, and it just seemed to me like reading about the service could make it sound like users have to do something special when all they want is forecasts on their dashboard.  It's probably me overthinking this.

MatthewFlamm replied:
I'm not sure this belongs in the NWS documentation, since it applies to _all_ integrations with weather entities.  I would look to see if you think the description in the weather integration itself needs to be clarified.

The00Dustin committed:
Further clarify weather entity purpose

Co-authored-by: MatthewFlamm <39341281+MatthewFlamm@users.noreply.github.com>

The00Dustin commented:
I committed the wrong one, I'll fix it and recommit in my branch.

The00Dustin committed:
Correct commit of wrong proposal

MatthewFlamm reviewed (approved):
Thanks!

The00Dustin commented:
I don't think I can remove tags, but do the in-progress (I think we are done) and need-rebase (I think this auto-tag was a mistake) tags need to be removed for this to get approved/merged?

c0ffeeca7 commented:
> I don't think I can remove tags, but do the in-progress (I think we are done) and need-rebase (I think this auto-tag was a mistake) tags need to be removed for this to get approved/merged?

Because it had a related core PR linked, it looked like it needed to target the `next` branch. However, because the parent core PR has already been merged in the last release, it can go onto `current`. For PRs with a merged related core PR there is the `parent-merged` label. I usually filter for those before reviewing because sometimes the core PRs are closed - making the docs review a waste of time.

c0ffeeca7 reviewed (approved):
Thank you, @The00Dustin :+1:

frenck commented on source/_integrations/nws.markdown:
```diff
@@ -25,6 +25,8 @@ According to the [API documentation](https://www.weather.gov/documentation/servi
 
 Details about the API are available in the [NWS API documentation](https://www.weather.gov/documentation/services-web-api). The [pynws](https://github.com/MatthewFlamm/pynws) library is used to retrieve data.
+
+Details about the `weather.get_forecasts` service are available in the [`weather` documentation](https://www.home-assistant.io/integrations/weather/).
```
Late review comment, there is a full link here, which shouldn't be used.

Please address this in a new PR to remove the `https://www.home-assistant.io` part from the target URL.

../Frenck

c0ffeeca7 replied:
Thank you for the catch! https://github.com/home-assistant/home-assistant.io/pull/32348
